src/common/src/util/value_encoding/column_aware_row_encoding.rs

```diff
@@ -105,8 +106,71 @@ impl RowEncoding {
     }
 }
 
+pub trait ValueRowSerializer: Clone {
```

Comment:
These traits and `BasicSerializer` seem not specific to or belong to "column-aware" row encoding. What about moving it to `mod.rs`? BTW, please add docs to all public structures. :)